pkg/api/types.go

```diff
@@ -799,6 +799,22 @@ type NodeCondition struct {
 	Message            string            `json:"message,omitempty"`
 }
 
+type NodeAddressKind string
+
+// These are valid address type of node. NodeLegacyHostIP is used to transit from out-dated
+// HostIP field to NodeAddress.
+const (
+	NodeLegacyHostIP NodeAddressKind = "LegacyHostIP"
+	NodeHostName     NodeAddressKind = "Hostname"
```

Comment:
What is the legacy HostIP? Internal IP? External IP? Hostname? It depends on the cloud provider?